tests/api.rs

```diff
@@ -136,6 +137,11 @@ impl<'a> Segments<'a> {
         Ok(sections)
     }
 }
+impl<'a> Default for Segments<'a> {
```

Comment:
Why is this needed?  You can simply add `#[derive(Default)]` on the `Segments` struct, and make `new()` call `Self::default()`.  Also, I am not too certain if both `default()` and `new()` are needed. I think cleaner interface would be to have just one. This way if later you need `new(some_init_value)`, it can be easily added.

Reply:
Good idea, I can get to this tomorrow.